PyInstaller/building/api.py

```diff
@@ -468,8 +469,9 @@ def __init__(self, *args, **kwargs):
             logger.warning('Ignoring hide_console; supported only on Windows!')
             self.hide_console = None
 
-        if self.contents_directory in ("", ".", "..") \
-                or "/" in self.contents_directory or "\\" in self.contents_directory:
+        if self.contents_directory in ("", "."):
+            self.contents_directory = None  # Re-enable old onedir layout without contents directory.
+        elif self.contents_directory == ".." or "/" in self.contents_directory or "\\" in self.contents_directory:
             raise SystemExit(
                 f'Invalid value "{self.contents_directory}" passed to `--contents-directory` or `contents_directory`. '
                 'Exactly one directory level is required.'
```

Comment:
Guess we'll need to mention `.` in the error message too

Reply:
You mean `"Exactly one directory level (or . to disable directory) is required."`?

Reply:
Yeah